iceoryx_posh/include/iceoryx_posh/roudi/roudi_app.hpp

```diff
@@ -47,8 +47,8 @@ class RouDiApp
 
   protected:
     /// @brief waits for the next signal to RouDi daemon
-    [[deprecated(
-        "use iox::posix::waitForTerminationRequest() from 'iceoryx_hoofs/posix_wrapper/signal_watcher.hpp'")]] bool
+    [[deprecated("removed in 4.0, use iox::posix::waitForTerminationRequest() from "
```

Comment:
I like this.

Disclaimer: I don't suggest to do this in this PR

It would be great to have this formalized and checked by the CI. A a script ninja could create a script to check for a specific syntax for our deprecation attributes. In combination with the `VERSION` file, this could tell us that there are deprecated functions for the new release and we need to remove them.

Do you think it might make sense to also tell the user which release introduce the deprecation? Just in case we have something that will be remove in next+2, e.g.
`[[deprecated("in 3.0, removed in 6.0, use foo::bar()")]]`